src/js/hooks/productSupplier/form/useProductSupplierForm.js

```diff
@@ -125,6 +111,7 @@ const useProductSupplierForm = () => {
       ratingTypeCode: values?.ratingTypeCode ? values.ratingTypeCode.id : null,
       productSupplierPreferences: values?.productSupplierPreferences,
       uom: values?.uom ? values.uom.id : null,
+      defaultPreferenceType: values?.defaultPreferenceType ? values.defaultPreferenceType.id : null,
```

Comment:
Is `values?.defaultPreferenceType?.id` not enough? I think that when the `values?.defaultPreferenceType` will be empty, then `id` should be `null` / `undefined`